src/core/keychain/RainbowSigner.ts

```diff
@@ -0,0 +1,93 @@
+import { TransactionFactory } from '@ethereumjs/tx';
+import { TransactionRequest } from '@ethersproject/abstract-provider';
+import { Signer } from '@ethersproject/abstract-signer';
+import { BigNumber } from '@ethersproject/bignumber';
+import { Bytes } from '@ethersproject/bytes';
+import { defineReadOnly } from '@ethersproject/properties';
+import { Provider } from '@ethersproject/providers';
+import { personalSign } from '@metamask/eth-sig-util';
+import { bytesToHex } from 'ethereum-cryptography/utils';
+import { Address } from 'wagmi';
+
+import { addHexPrefix } from '../utils/hex';
+
+import { PrivateKey } from './IKeychain';
+
+export class RainbowSigner extends Signer {
+  readonly privateKey!: PrivateKey;
+  readonly address!: Address;
+
+  constructor(provider: Provider, privateKey: PrivateKey, address: Address) {
+    super();
+    defineReadOnly(this, 'provider', provider);
+    defineReadOnly(this, 'privateKey', privateKey);
+    defineReadOnly(this, 'address', address);
+  }
+
+  #getPrivateKeyBuffer = (): Buffer => {
+    return Buffer.from(addHexPrefix(this.privateKey).substring(2), 'hex');
+  };
+
+  async getAddress(): Promise<string> {
+    return this.address as string;
+  }
+
+  async signMessage(message: Bytes | string): Promise<string> {
+    const data =
+      typeof message === 'string' ? message : bytesToHex(message as Uint8Array);
+    const pkey = this.#getPrivateKeyBuffer();
+
+    const signature = personalSign({
+      privateKey: pkey,
+      data,
+    });
+    return signature;
+  }
+
+  async signTransaction(transaction: TransactionRequest): Promise<string> {
+    // We're converting the ethers v5 transaction request to
+    // an ethereum JS transaction object so all the crypto operations
+    // are made using EthereumJS instead of ethers v5
+
+    const typedTx = TransactionFactory.fromTxData({
+      data: transaction.data?.toString(),
+      to: transaction.to,
+      accessList: [],
+      maxPriorityFeePerGas: transaction.maxPriorityFeePerGas
+        ? BigNumber.from(transaction.maxPriorityFeePerGas).toHexString()
+        : undefined,
+      maxFeePerGas: transaction.maxFeePerGas
```

Comment:
no gasPrice for legacy supported chains?

Reply:
good call. missed that one. Just added it